### ISSUE TITLE
Question: Configuration for 50+ switches?

### ISSUE THREAD

**Issue by dR3b** [sensu/sensu#1278]
Hi

I have a default "sensu + uchiwa" installation running. The sensu-server and the
sensu-client are on the same VM.

I also added a simple "ping" check for 1 switch:

```
{
  "checks": {
    "sw1": {
      "command": "/opt/sensu/embedded/bin/check-ping.rb -h sw1",
      "subscribers": [
        "sw"
      ],
      "interval": 30
    }
  }
}
```

This is working with no problem. Now i want to add 50 more switches. Do I really have to repeate this step for every switch?

Thanks

**Comment by kobbe** [comment]
Since commands can be any command it is quite easy to just write/script your own checks. Make for example a python/ruby/bash script that loops through all desired hosts.

I usually do it like this in linux:

```#!/bin/bash

#This program returns the greatest exit status

EXIT_STATUS=0
echo -n "Description of what the output is: "; COMMAND || EXIT_STATUS=$(($EXIT_STATUS>$? ? $EXIT_STATUS : $?))
echo -n "Description of what the output 2 is: "; COMMAND2 || EXIT_STATUS=$(($EXIT_STATUS>$? ? $EXIT_STATUS : $?))
exit $EXIT_STATUS
...
```

If you don't want to manually add all rows you can just make a loop.

**Comment by dR3b** [comment]
Sorry, but that sounds terrible for me. That's not a solution that I expect for a "Monitoring for today's infrastructure" tool.

I need a "virtual" host for every switch. I think something like this:
```
{
  "checks": {
    "::SWITCHNAME::": ["sw1", "sw2", "sw3", "swN"],
      "source": "::SWITCHNAME::",
      "command": "/opt/sensu/embedded/bin/check-ping.rb -h ::SWITCHNAME::",
      "subscribers": ["sw"],
      "interval": 30
    }
  }
}
```
One check definition for all switches (sw1, sw2, swN).

**Comment by yanc0** [comment]
There is no concept of distant check in sensu as far I know. All machines that you want to be monitored have to run the `sensu-client` daemon. It seems that sensu is not the right tool for network assets monitoring. I figured out distant monitoring with a little HTTP Client As a Service that I rapidly developed, It is called [pingmeback](https://github.com/yanc0/pingmeback) with a sensu plugin named [sensu-plugins-pingmeback](https://github.com/yanc0/sensu-plugins-pingmeback) for distant monitoring purposes. Maybe this can be a begin of an answer.

I was in the same situation and I am very interested by the next answers in this issue.

**Comment by kobbe** [comment]
I think It definitely could be something that sensu could implement. It would certainly remove some of the scripting from sensu in some scenarios.

**Comment by dR3b** [comment]
@yanc0 
Thank you. I think without a "distant check" concept sensu will never replace `nagios` or `icinga`. I'm really sorry, but at this stage `sensu` for me is useless. I need it for network stuff. For "software" monitoring we already use `prometheus`.

**Comment by calebhailey** [comment]
@dR3b thank you for submitting this issue! 

We definitely don't disagree that we need a better solution for configuring checks to be executed against proxy clients (i.e. external systems, including network devices, etc). We have started to work out some possible solutions to this, and we'll publish our proposals as GitHub issues as soon as they are solidified. Unfortunately, I think it may be a few more weeks before we get there, and at least 6-8 weeks before an initial implementation would be available (if at all).

**Comment by dR3b** [comment]
@calebhailey Thank you!

**Comment by dR3b** [comment]
Any news?

**Comment by x70b1** [comment]
Same problem here! A solution for a more simple config would be nice.

**Comment by jflemer-ndp** [comment]
Looks like a dup of #1259.

**Comment by cwjohnston** [comment]
In #1555 we're proposing to implement "proxy check requests", a new feature allowing a pubsub check definition to generate a unique check request for each client matching a defined set of client attributes.

Please have a look and share your thoughts?

**Comment by portertech** [comment]
We're very excited by Proxy Check Requests, now available in the Sensu 0.28 release(s), which can dynamically generate proxy check requests for clients (and proxy clients) in the registry.

I look forward to the ability to manage the Sensu client registry via the Uchiwa dashboard :tada: 

Closing this issue due to inactivity following @cwjohnston's update.

**Comment by jflemer-ndp** [comment]
Yay!
